snapshot/bundlestore/http_server.go

```diff
@@ -79,6 +79,28 @@ func (s *httpServer) HandleUpload(w http.ResponseWriter, req *http.Request) {
 	s.storeConfig.Stat.Counter(stats.BundlestoreUploadOkCounter).Inc(1)
 }
 
+func (s *httpServer) CheckExistence(w http.ResponseWriter, req *http.Request) {
+	log.Infof("Checking Existence %v %v (from %v)", req.Host, req.URL, req.RemoteAddr)
+	defer s.storeConfig.Stat.Latency(stats.BundlestoreCheckLatency_ms).Time().Stop()
+	s.storeConfig.Stat.Counter(stats.BundlestoreCheckCounter).Inc(1)
+	bundleName := strings.TrimPrefix(req.URL.Path, "/bundle/")
+	if err := checkBundleName(bundleName); err != nil {
+		log.Infof("Bundlename err: %v --> StatusBadRequest (from %v)", err, req.RemoteAddr)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		s.storeConfig.Stat.Counter(stats.BundlestoreCheckErrCounter).Inc(1)
+		return
+	}
+	ok, err := s.storeConfig.Store.Exists(bundleName)
+	if err != nil || !ok {
+		log.Infof("Check err: %v --> StatusNotFound (from %v)", err, req.RemoteAddr)
+		http.NotFound(w, req)
+		s.storeConfig.Stat.Counter(stats.BundlestoreCheckErrCounter).Inc(1)
+		return
+	}
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
```

Comment:
why set this header?

Reply:
To match the GET response, the http lib adds this automatically for GETs.